### PR TITLE
epic: esl-trigger for v3.0.0

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

File: src/modules/esl-trigger/core/esl-trigger.ts [hunk 1]
```diff
@@ -101,6 +70,28 @@ export class ESLTrigger extends ESLBaseElement {
...
+  }
+  /** Show delay attribute processing */
+  public get showDelayValue(): number | undefined {
+    return ESLMediaRuleList.parse(this.showDelay, ESLTrigger.parseDelayValue).activeValue;
+  }
+  /** Hide delay attribute processing */
+  public get hideDelayValue(): number | undefined {
+    const hideDelay = +(this.hideDelay || 'none');
+    return isNaN(hideDelay) ? undefined : hideDelay;
+  }
```
Comment (Sisha3342) [inline src/modules/esl-trigger/core/esl-trigger.ts:93]: Maybe we can use ESLTrigger.parseDelayValue(this.hideDelay || 'none') here?

**Review by Sisha3342 — APPROVED** [review]

File: src/modules/esl-trigger/core/esl-trigger.ts (hunk 1 above)
Comment (ala-n) [reply src/modules/esl-trigger/core/esl-trigger.ts:93]: Yea, good catch. Thanks!

**Review by dshovchko — APPROVED** [review]

**Comment by ala-n** [comment]
@dshovchko , @Sisha3342 , sorry that adding changes above.  Please check [small a11y additions](https://github.com/exadel-inc/esl/pull/243/commits/49e30c458e6860955718440451fe91ee7545fdc1) in bounds of that PR.

**Review by Sisha3342 — APPROVED** [review]

**Review by andreybelous — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-trigger/core/esl-trigger.ts [hunk 2]
```diff
@@ -200,15 +177,53 @@ export class ESLTrigger extends ESLBaseElement {
...
   }
 
+  /** Handles `click` event */
+  @bind
+  protected _onClick(event: MouseEvent) {
+    if (!this.allowClick || this.isIgnoredTarget(event.target)) return;
+    event.preventDefault();
+    switch (this.mode) {
+      case 'show': return this.showTarget({event});
+      case 'hide': return this.toggleTarget({event});
```
Comment (ala-n) [inline src/modules/esl-trigger/core/esl-trigger.ts:187]: hideTarget

File: src/modules/esl-trigger/core/esl-trigger.ts [hunk 3]
```diff
@@ -200,15 +177,53 @@ export class ESLTrigger extends ESLBaseElement {
...
-      switch (this.mode) {
-        case 'show': return this._onShowEvent(event);
-        case 'hide': return this._onHideEvent(event);
-        default: return this._onToggleEvent(event);
-      }
+    if (![ENTER, SPACE].includes(event.key) || this.isIgnoredTarget(event.target)) return;
+    event.preventDefault();
+    switch (this.mode) {
+      case 'show': return this.showTarget({event});
+      case 'hide': return this.toggleTarget({event});
```
Comment (ala-n) [inline src/modules/esl-trigger/core/esl-trigger.ts:199]: hideTarget

File: src/modules/esl-trigger/core/esl-trigger.ts [hunk 4]
```diff
@@ -30,57 +30,26 @@ export class ESLTrigger extends ESLBaseElement {
 
   /** Target Toggleable {@link TraversingQuery} selector. `next` by default */
   @attr({defaultValue: 'next'}) public target: string;
-  /** Event to handle by trigger. Support `click`, `hover` modes or any custom. `click` by default */
-  @attr({defaultValue: 'click'}) public event: string;
   /** Action to pass to the Toggleable. Supports `show`, `hide` and `toggle` values. `toggle` by default */
   @attr({defaultValue: 'toggle'}) public mode: string;
 
+  @attr({defaultValue: 'all'}) public trackClick: string;
```
Comment (ala-n) [inline src/modules/esl-trigger/core/esl-trigger.ts:36]: JSDoc

**Review by abarmina — APPROVED** [review]

**Pull request merged**